lib/mayaUsd/undo/UsdUndoStateDelegate.cpp

```diff
@@ -22,6 +22,42 @@
 
 PXR_NAMESPACE_USING_DIRECTIVE
 
+namespace {
+void copySpecAtPath(const SdfAbstractData& src, SdfAbstractData* dst, const SdfPath& path)
+{
+    dst->CreateSpec(path, src.GetSpecType(path));
```

Comment:
I don't think we need it since _dst is basically a pointer to a layer's internal data but added it anyways.

```
auto layerDataPtr = get_pointer(_GetLayerData());
TF_AXIOM(layerDataPtr);
SpecCopier specCopier(layerDataPtr);
```

_GetLayerData function definition:
```
SdfAbstractDataPtr 
SdfLayerStateDelegateBase::_GetLayerData() const
{
    return _layer ? SdfAbstractDataPtr(_layer->_data) : SdfAbstractDataPtr();
}
```